code/game/objects/structures/signs/_signs.dm

```diff
@@ -187,12 +187,12 @@
 
 /obj/item/sign/add_context(atom/source, list/context, obj/item/held_item, mob/user)
 	. = ..()
-	if(is_editable && istype(held_item, /obj/item/pen))
+	if(is_editable && held_item.get_writing_implement_details()?["interaction_mode"] == MODE_WRITING)
```

Comment:
Be careful, you changed some checks like this from something null-safe to something that will runtime

`istype(null, pen)` works, this does not. 

Add context is namely what you should be worried about, attackby and co are safe